### PR TITLE
Chore: use errutil for pluginRepo errors

### PR DESCRIPTION
**Pull request by gamab**
**What is this feature?**

At some point while working on https://github.com/grafana/grafana/pull/78211 I rewrote the `pluginrepo`'s error to use `errutil` (since i had to do the same error check for my function fetching the plugin.json from grafana.com).
Now there isn't much value in it since we removed this extra call.  But in case this is something you want I figured let's not loose that work. If it's not something you want, just close the PR :slightly_smiling_face: 

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/plugins/repo/errors.go [hunk 1]
```diff
@@ -43,47 +47,44 @@ func (e ErrResponse4xx) Error() string {
...
-}
+	ErrArcNotFoundMsg  = "{{.Public.PluginID}} is not compatible with your system architecture: {{.Public.SysInfo}}"
+	ErrArcNotFoundBase = errutil.NewBase(errutil.StatusNotFound, "plugin.archNotFound").
+				MustTemplate(ErrArcNotFoundMsg, errutil.WithPublic(ErrArcNotFoundMsg))
 
-func (e ErrVersionNotFound) Error() string {
-	return fmt.Sprintf("%s v%s either does not exist or is not supported on your system (%s)", e.pluginID, e.requestedVersion, e.systemInfo)
-}
+	ErrChecksumMismatchMsg  = "expected SHA256 checksum does not match the downloaded archive ({{.Public.ArchiveURL}}) - please contact security@grafana.com"
+	ErrChecksumMismatchBase = errutil.NewBase(errutil.StatusInternal, "plugin.checksumMismatch").
```
Comment (gamab) [inline pkg/plugins/repo/errors.go:64]: Maybe 422 would be better suited.

**Review by andresmgot — COMMENTED** [review]
It makes sense to me, letting @marefr to do an additional review since he's been working more with errors

File: pkg/util/errutil/status.go [hunk 2]
```diff
@@ -20,6 +20,10 @@ const (
 	// corresponding document to return to the request.
 	// HTTP status code 404.
 	StatusNotFound CoreStatus = "Not found"
+	// StatusConflict means that the server cannot fulfilling the request
```
Comment (gamab) [inline pkg/util/errutil/status.go:23]:
```suggestion
	// StatusConflict means that the server cannot fulfill the request
```

File: pkg/plugins/repo/errors.go [hunk 3]
```diff
@@ -43,47 +47,44 @@ func (e ErrResponse4xx) Error() string {
...
+var (
+	ErrVersionUnsupportedMsg  = "{{.Public.PluginID}} v{{.Public.Version}} is not supported on your system {{.Public.SysInfo}}"
+	ErrVersionUnsupportedBase = errutil.NewBase(errutil.StatusConflict, "plugin.unsupportedVersion").
+					MustTemplate(ErrVersionUnsupportedMsg, errutil.WithPublic(ErrVersionUnsupportedMsg))
 
-func (e ErrVersionUnsupported) Error() string {
-	return fmt.Sprintf("%s v%s is not supported on your system (%s)", e.pluginID, e.requestedVersion, e.systemInfo)
-}
+	ErrVersionNotFoundMsg  = "{{.Public.PluginID}} v{{.Public.Version}} either does not exist or is not supported on your system {{.Public.SysInfo}}"
+	ErrVersionNotFoundBase = errutil.NewBase(errutil.StatusNotFound, "plugin.versionNotFound").
```
Comment (marefr) [inline pkg/plugins/repo/errors.go:56]:
```suggestion
	ErrVersionNotFoundBase = errutil.NotFound("plugin.versionNotFound").
```

**Review by marefr — CHANGES_REQUESTED** [review]
Looks great. Just some minor suggestions

**Review by marefr — APPROVED** [review]
LGTM

**Pull request merged**